### PR TITLE
Fully automate learner setup with Gitpod

### PR DESCRIPTION
**Pull request by jacobparis**
After playing around with the [Remix blog tutorial](https://remix.run/docs/en/v1/tutorials/blog#prerequisites) in Gitpod, I wanted to see how much the Epic React learner process could be simplified the same way.

Looking at the issues in this repository, most of them are about local dev environment issues, whether they're incompatible Node versions or Windows bugs with the setup scripts. More than half the README.md (all-contributors excluded) is about navigating setup issues too.

After adding the `gitpod.yml` file, from the time I arrived at the repository, I was able to set up the application in my local VS Code, add my email to the questionnaires, run the tests, and complete the first exercise in under 2 minutes.

[Running Epic React Fundamentals in Gitpod](http://www.youtube.com/watch?v=22Rkzyqt5ao "Running Epic React Fundamentals in Gitpod")

Everyone who runs this repo in Gitpod can bring their own local VS Code or JetBrains IDE, with all their local settings, but they don't have to worry about having git, node, or npm installed, and the setup scripts will work for windows users too because they're just connecting to a linux environment that does all the work for them.

There is already a CodeSandbox link set up, which solves a similar problem although it only works in the browser and doesn't support the "Set your email" setup script. Gitpod **can** work in the browser too, but I think it's best to learn new things with the same tools and IDEs you do your regular development in.

You can try this out yourself to see how the workflow feels

[Open in Gitpod](https://gitpod.io/#https://github.com/jacobparis/react-fundamentals)

I'm not sure if you still intend on running any more Epic React live workshops, but eliminating these sorts of setup difficulties works [really well in live workshops](https://www.gitpod.io/blog/workshops-as-code) and I would love to work through the rest of this course to integrate Epic React

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (kentcdodds) [inline .vscode/settings.json:1]: Can we put this in the `kcd.settings.json`? I don't like surprising people with settings, but they can copy them if they want them.

**Review by kentcdodds — CHANGES_REQUESTED** [review]
Thanks! This seems great to me 👍 Just one thing.

File: .vscode/settings.json (hunk 1 above)
Comment (jacobparis) [reply .vscode/settings.json:1]: Oh yes, I think that's a good call

Comment (jacobparis) [reply .vscode/settings.json]: Confirmed they have been copied over and the un-suffixed `settings.json` has been removed

**Review by kentcdodds — APPROVED** [review]
Thanks!

**Pull request merged**